### ISSUE TITLE
Blocks with lower height then the current best tip should be passed to consensus

### ISSUE THREAD

**Issue by akoptelov** [openmina/openmina#35]
Now there is the enabling condition that prevents blocks with lower height to be passed to the consensus handling:
https://github.com/openmina/openmina/blob/0e0039bfdf8ae1942b029cf98740129e7847efa8/node/lib/src/consensus/consensus_actions.rs#LL44C1-L45C84

This is not correct. If an adversary forges its own chain, eventually it might get more and more stakes, having higher block density, and higher chain height. In this case, if the node for some reason first connects to such malicious chain, it will be locked there, despite the fact that its minimal density is less than of the main chain.

We might use global slots here, to avoid handling blocks from the past, but that is not accurate as well, as slots might be skipped in the better chain.

@binier FYI

**Comment by binier** [comment]
True. Also we should do consensus check first and then block proof verification, as consensus check is much faster.
